backend/catalogue/management/commands/generate_pressure_previews.py

```diff
@@ -0,0 +1,161 @@
+from django.core.management.base import BaseCommand
+from django.core.files.storage import default_storage
+from django.core.files import File
+from django.conf import settings
+from urllib.request import urlopen
+from urllib.parse import urlencode
+from PIL import Image
+from io import BytesIO
+import logging
+
+from catalogue.models import KeyedLayer, RegionReport, Pressure
+
+Image.MAX_IMAGE_PIXELS = None
+
+basemap = Image.open(default_storage.open('land_shallow_topo_21600.tif'))
+
+
+def basemap_latitude_to_pixel(latitude):
+    t = 0.5 + (latitude / 360)
+    return round(basemap.width * t)
+
+
+def basemap_longitude_to_pixel(longitude):
+    t = 0.5 - (longitude / 180)
+    return round(basemap.height * t)
+
+
+def basemap_bbox(north, south, east, west):
+    left = basemap_latitude_to_pixel(west)
+    right = basemap_latitude_to_pixel(east)
+    upper = basemap_longitude_to_pixel(north)
+    lower = basemap_longitude_to_pixel(south)
+
+    if (left > right):
+        left_img = basemap.crop((left, upper, basemap.width + right, lower))
+        right_img = basemap.crop((0, upper, right, lower))
+        left_img.paste(right_img, (left_img.width - right, 0))
+        return left_img
+    else:
+        return basemap.crop((left, upper, right, lower))
+
+def get_layer_image(layer, bbox, size):
+    sub_params = {
+        'service': 'WMS',
+        'request': 'GetMap',
+        'layers': layer.layer_name,
+        'styles': layer.style or '',
+        'format': 'image/png',
+        'transparent': 'true',
+        'version': '1.1.1',
+        'width': size['width'],
+        'height': size['height'],
+        'srs': 'EPSG:4326',
+        'bbox': '{west},{south},{east},{north}'.format(**bbox)
+    }
+
+    url = f'{layer.server_url}?{urlencode(sub_params)}'
+    logging.info(f'Retrieving layer image from: {url}')
+    image = Image.open(urlopen(url)).convert('RGBA')
+    logging.info('Layer image retrieval comple')
+    return image
+
+def get_boundary_layer_image(layer, bbox, size, network, park):
+    sub_params = {
+        'service': 'WMS',
+        'request': 'GetMap',
+        'layers': layer.layer_name,
+        'styles': layer.style or '',
+        'format': 'image/png',
+        'transparent': 'true',
+        'version': '1.1.1',
+        'width': size['width'],
+        'height': size['height'],
+        'srs': 'EPSG:4326',
+        'bbox': '{west},{south},{east},{north}'.format(**bbox),
+        'cql_filter': f'RESNAME=\'{park}\'' if park else f'NETNAME=\'{network}\''
```

Comment:
Minor: you can use f-strings with double-quotes here, which would avoid the need to escape the internal single-quotes.  Might look out of place with the rest of the entries though?  (not sure what the single/double preference is in pep8 actually; I assume they have opinions)